### PR TITLE
Fix a bug with moderator dropdowns and add aria tags

### LINKED ISSUE #13318: `dropdown-options` always open and lacks a11y
**Describe the bug**
Drop down options in for items in the "OTHER THINGS YOU CAN DO" section of the mod panel are always open.
While investigating the cause I noticed there were no aria attributes

ref: <https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html>

**To Reproduce**
- Go to a post
- Open the mod panel
- See the accordion appears open
- Click the accordion and see nothing changes

**Expected behaviour**
The accordion should open and close on click as well as modify it's aria attributes

**Screenshots**
dev__dropdown-options

**Desktop (please complete the following information):**

- OS, version: Windows 10
- Browser, version: 89.0.4389.114

**Smartphone (please complete the following information):**
N/A

**Additional context**
The `display: flex` here 
https://github.com/forem/forem/blob/009c14653ab84da91cf278400350503e056bef29/app/assets/stylesheets/views/mod-actions.scss#L367-L372

Overrides the `display: none` generated here
https://github.com/forem/forem/blob/009c14653ab84da91cf278400350503e056bef29/app/assets/stylesheets/config/_generator.scss#L600

**Solutions:** __for visible bug__
- add hidden to `mod-actions` CSS for higher specificity
- create crayons-accordion that solves all these issues
- use summary/details instead of doing JS magic

### PR DESCRIPTION
**Pull request by Link2Twenty**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Fix a bug with moderator dropdowns and add aria tags.

## Related Tickets & Documents
closes #13318

## QA Instructions, Screenshots, Recordings
forem_mod_dropdown--fix

### UI accessibility concerns?
I'm not really sure what this is meant to do, it looks like it just greys out other dropdowns unless they're open but then still allows click actions to work.

https://github.com/forem/forem/blob/c842d1a0c6718f319f167e9ec13d4d7d5d6b6d39/app/javascript/actionsPanel/actionsPanel.js#L371-L380

## Added tests?

- [ ] Yes
- [x] No, and this is why: old tests will still work
- [ ] I need help with writing tests

## [optional] Are there any post deployment tasks we need to perform?
N/A

## [optional] What gif best describes this PR or how it makes you feel?

slowing pushing things over

### CODE REVIEW

**Comment by Link2Twenty** [comment]
Didn't realise this was connected to #12297
I've now added `aria-haspopup` as is indicated in that issue.

File: app/assets/stylesheets/views/mod-actions.scss [hunk 1]
```diff
@@ -369,6 +369,10 @@
       flex-direction: column;
       padding: 0 var(--su-4);
       transition-duration: 1s;
+
+      &.hidden {
```
Comment (nickytonline) [inline app/assets/stylesheets/views/mod-actions.scss:373]: Refactor (blocking): We have a CSS utility class [hidden](https://storybook.forem.com/?path=/story/utility-first-classes-display--hidden).

File: app/javascript/actionsPanel/actionsPanel.js [hunk 2]
```diff
@@ -258,9 +253,8 @@ async function adjustTag(el) {
 
       // eslint-disable-next-line no-restricted-globals
       top.addSnackbarItem({
-        message: `The #${adjustedTagName} tag was ${
-          outcome.result === 'addition' ? 'added' : 'removed'
-        }.`,
+        message: `The #${adjustedTagName} tag was ${outcome.result === 'addition' ? 'added' : 'removed'
```
Comment (nickytonline) [inline app/javascript/actionsPanel/actionsPanel.js:256]: Prettier in action?

**Review by nickytonline — CHANGES_REQUESTED** [review]
Thanks as always @Link2Twenty for your contributions. Small change request.

File: app/assets/stylesheets/views/mod-actions.scss (hunk 1 above)
Comment (Link2Twenty) [reply app/assets/stylesheets/views/mod-actions.scss:373]: We do but it's wasn't specific enough so was being overwritten.

File: app/javascript/actionsPanel/actionsPanel.js (hunk 2 above)
Comment (Link2Twenty) [reply app/javascript/actionsPanel/actionsPanel.js]: Yep, I've reverted.

File: app/assets/stylesheets/views/mod-actions.scss (hunk 1 above)
Comment (Link2Twenty) [reply app/assets/stylesheets/views/mod-actions.scss]: OK,  I've reworked it to use a `not` in order to keep specify in check.

**Comment by aitchiss** [comment]
> I'm not really sure what this is meant to do, it looks like it just greys out other dropdowns unless they're open but then still allows click actions to work.

I think you're right 🤔  It's strange but I guess _maybe_ it helps visually to understand which dropdown is currently open? I'm not convinced though 😆

**Review by aitchiss — APPROVED** [review]
Looks good to me! Thanks for this. I tested locally, including with Safari/VoiceOver and the expanded/collapsed states are now being announced too ✨

File: app/javascript/actionsPanel/actionsPanel.js (hunk 2 above)
Comment (nickytonline) [reply app/javascript/actionsPanel/actionsPanel.js]: Oh no biggie. It was more a question. 😄

**Review by nickytonline — APPROVED** [review]
Thanks for making the changes @Link2Twenty and thanks so much for making Forem more accessible! 🔥

**Pull request merged**